cmake/partition_manager.cmake

```diff
@@ -27,231 +21,298 @@ macro(add_region)
-    add_region(
-      NAME external_flash
-      SIZE ${CONFIG_PM_EXTERNAL_FLASH_SIZE}
-      BASE ${CONFIG_PM_EXTERNAL_FLASH_BASE}
-      PLACEMENT start_to_end
-      DEVICE ${CONFIG_PM_EXTERNAL_FLASH_DEV_NAME}
-      )
-  endif()
+math(EXPR flash_size "${CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
 
-  set(pm_cmd
-    ${PYTHON_EXECUTABLE}
-    ${NRF_DIR}/scripts/partition_manager.py
-    --input-files ${input_files}
-    --regions ${regions}
-    --output-partitions ${CMAKE_BINARY_DIR}/partitions.yml
-    --output-regions ${CMAKE_BINARY_DIR}/regions.yml
-    ${static_configuration}
-    ${region_arguments}
-    )
+add_region(
+  NAME flash_primary
+  SIZE ${flash_size}
+  BASE ${CONFIG_FLASH_BASE_ADDRESS}
+  PLACEMENT complex
+  DEVICE NRF_FLASH_DRV_NAME
+  )
 
-  set(pm_output_cmd
-    ${PYTHON_EXECUTABLE}
-    ${NRF_DIR}/scripts/partition_manager_output.py
-    --input-partitions ${CMAKE_BINARY_DIR}/partitions.yml
-    --input-regions ${CMAKE_BINARY_DIR}/regions.yml
-    --config-file ${CMAKE_BINARY_DIR}/pm.config
-    --input-names ${images}
-    --header-files
-    ${header_files}
+if (CONFIG_PM_EXTERNAL_FLASH)
+  add_region(
+    NAME external_flash
+    SIZE ${CONFIG_PM_EXTERNAL_FLASH_SIZE}
+    BASE ${CONFIG_PM_EXTERNAL_FLASH_BASE}
+    PLACEMENT start_to_end
+    DEVICE ${CONFIG_PM_EXTERNAL_FLASH_DEV_NAME}
     )
+endif()
 
-  # Run the partition manager algorithm.
-  execute_process(
-    COMMAND
-    ${pm_cmd}
-    RESULT_VARIABLE ret
-    )
+set(pm_out_partition_files ${CMAKE_BINARY_DIR}/partitions_${domain}.yml)
+set(pm_out_region_files ${CMAKE_BINARY_DIR}/regions_${domain}.yml)
+set(pm_out_dotconf_files ${CMAKE_BINARY_DIR}/pm_${domain}.config)
+
+set(pm_cmd
+  ${PYTHON_EXECUTABLE}
+  ${NRF_DIR}/scripts/partition_manager.py
+  --input-files ${input_files}
+  --regions ${regions}
+  --output-partitions ${pm_out_partition_files}
+  --output-regions ${pm_out_region_files}
+  ${dynamic_partition_argument}
+  ${static_configuration}
+  ${region_arguments}
+  )
+
+set(pm_output_cmd
+  ${PYTHON_EXECUTABLE}
+  ${NRF_DIR}/scripts/partition_manager_output.py
+  --input-partitions ${pm_out_partition_files}
+  --input-regions ${pm_out_region_files}
+  --config-file ${pm_out_dotconf_files}
+  )
+
+# Run the partition manager algorithm.
+execute_process(
+  COMMAND
+  ${pm_cmd}
+  RESULT_VARIABLE ret
+  )
+
+if(NOT ${ret} EQUAL "0")
+  message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
+endif()
 
-  if(NOT ${ret} EQUAL "0")
-    message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
-  endif()
+# Produce header files and config file.
+execute_process(
+  COMMAND
+  ${pm_output_cmd}
+  RESULT_VARIABLE ret
+  )
 
-  # Produce header files and config file.
-  execute_process(
-    COMMAND
-    ${pm_output_cmd}
-    RESULT_VARIABLE ret
-    )
+if(NOT ${ret} EQUAL "0")
+  message(FATAL_ERROR "Partition Manager output generation failed, aborting. Command: ${pm_output_cmd}")
```

Comment:
Help the user by printing the error message from `execute_process`

Reply:
See https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2070#discussion_r411910900